examples/dojo/js/todo/computed.js

```diff
@@ -0,0 +1,161 @@
+	 * @example
+	 * <caption>
+	 *     If stateful.first is "John" and stateful.last is "Doe", stateful.name becomes "John Doe".
+	 * </caption>
+	 * computed(stateful, 'name', function(first, last){
+	 *     return first + ' ' + last;
+	 * }, at(stateful, 'first'), at(stateful, 'last'));
+	 * @example
+	 * <caption>
+	 *     If names is an array of objects with name property,
+	 *     stateful.totalNameLength becomes the sum of length of name property of each array item.
+	 * </caption>
+	 * computed(stateful, 'totalNameLength', function(names){
+	 *     var total = 0;
+	 *     array.forEach(names, function(name){
+	 *         total += name.length;
+	 *     });
+	 *     return total;
+	 * }, lang.mixin(at(names, 'name'), {each: true}));
+	 */
+	return function (target, targetProp, compute) {
+		function applyComputed(data) {
+			var result;
+			var hasResult;
+
+			try {
+				result = compute.apply(target, data);
+				hasResult = true;
+			} catch (e) {
+				console.error('Error during computed property callback: ' + (e && e.stack || e));
+			}
+
+			if (hasResult) {
+				if (typeof target.set === 'function') {
+					target.set(targetProp, result);
+				} else {
+					target[targetProp] = result;
+				}
+			}
+		}
+
+		if (target === null || target === undefined) {
+			throw new Error('Computed property cannot be applied to null.');
+		}
+		if (targetProp === '*') {
+			throw new Error('Wildcard property cannot be used for computed properties.');
+		}
+
+		var deps = arrayProto.slice.call(arguments, 3);
+		var hDep = array.map(deps, function (dep, index) {
+			function observeEntry(entry) {
+				return watch(entry, dep.targetProp, function () {
+					applyComputed(getProps(deps));
+				});
+			}
+
+			if (dep.targetProp === '*') {
+				throw new Error('Wildcard property cannot be used for computed properties.');
+			} else if (dep.each) {
+				var hArray;
+				var hEntry = array.map(dep.target, observeEntry);
+
+				if (dep.target && typeof dep.target.watchElements === 'function') {
+					hArray = dep.target.watchElements(function (idx, removals, adds) {
+						removeHandles(arrayProto.splice.apply(hEntry, [idx, removals.length].concat(array.map(adds, observeEntry))));
+						applyComputed(getProps(deps));
+					});
+				} else {
+					console.log('Attempt to observe non-stateful-array ' + dep.target + '. Observation not happening.');
+				}
+
+				return {
+					remove: function () {
+						if (hArray) {
+							hArray.remove();
+						}
+						removeHandles(hEntry);
+					}
+				};
+			} else {
+				return watch(dep.target, dep.targetProp, function (current) {
+					var list = [];
+					arrayProto.push.apply(list, getProps(deps.slice(0, index)));
+					list.push(current);
+					arrayProto.push.apply(list, getProps(deps.slice(index + 1)));
+					applyComputed(list);
+				});
+			}
+		});
+
+		applyComputed(getProps(deps));
+		return {
+			remove: function () {
+				removeHandles(hDep);
+			}
+		};
+	};
+});
```

Comment:
That is a very old discussion though. I just feel like having to manually reinvent something like that is a big Dojo turn-off for anyone looking at this app.